hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/SeqIdGenInfoHandler.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.tasks;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.ozone.recon.scm.ReconScmMetadataManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Manages sequenceId Table info of SCM MetaData DB, updates Recon's in-memory
+ * SequenceIdGenerator objects.
+ */
+public class SeqIdGenInfoHandler implements SCMMetaDataTableHandler {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(SeqIdGenInfoHandler.class);
+
+  private ReconScmMetadataManager scmMetadataManager;
+
+  public SeqIdGenInfoHandler(ReconScmMetadataManager scmMetadataManager) {
+    this.scmMetadataManager = scmMetadataManager;
+  }
+
+  /**
+   * Handles a PUT event on scm metadata DB tables.
+   *
+   * @param event The PUT event to be processed.
+   */
+  @Override
+  public void handlePutEvent(RocksDBUpdateEvent<?, Object> event) {
+    final String sequenceIdName = (String) event.getKey();
+    final Long lastId = (Long) event.getValue();
+    try {
+      LOG.info("handlePutEvent for add new seqIdGen event for sequence name: {}", sequenceIdName);
+      handlePutSequenceIdGenEvent(sequenceIdName, lastId);
+    } catch (IOException ioe) {
+      LOG.error("Unexpected error while handling add new seqIdGen event for sequence name: {} - ", sequenceIdName, ioe);
+    }
+  }
+
+  private void handlePutSequenceIdGenEvent(String sequenceIdName, Long lastId) throws IOException {
+    scmMetadataManager.getSequenceIdGen().initialize(sequenceIdName, lastId);
+  }
+
+  /**
+   * Handles a DELETE event on scm metadata DB tables.
+   *
+   * @param event The DELETE event to be processed.
+   */
+  @Override
+  public void handleDeleteEvent(RocksDBUpdateEvent<?, Object> event) {
+    final String sequenceIdName = (String) event.getKey();
+    LOG.error("Delete event should not be raised on sequenceId table for sequence name: {} - ", sequenceIdName);
```

Comment:
We can utilise `UnsupportedOperationException` when the method is not permitted to execute, instead of logging an error that might mislead due to an unsupported operation?

Reply:
This delete event, we can throw `UnsupportedOperationException`. Will do change.